### ISSUE TITLE
How to check if request limit is exceed

### ISSUE THREAD

**Issue by sjmohanta** [LordMike/TMDbLib#240]
Sometimes whiles requesting for a movie information from production i am getting error but while trying to request same from development i donot get this issue.

The above issue also gets resolved after some times. So, i suspect that it has t do with the request limit which is 40 requests/10 sec. It would be sufficient for my usage, also some times it takes hours to return data.

I want to know the rate limit parameter value as given in the below link.
https://developers.themoviedb.org/3/getting-started/request-rate-limiting

How can i detect is this error is due to rate limit. I also tried to look into code base but was not able to get it.
Your help in above would be appriciated.
Thanks

**Comment by LordMike** [comment]
You can see the current ratelimit-handling code here:
https://github.com/LordMike/TMDbLib/blob/fec9e03945343c9f2440cc965488ebda12a0d32c/TMDbLib/Rest/RestRequest.cs#L216-L228

The library should try to resolve rate limits by retrying stuff up to `MaxRetryCount` times. If it fails, you'll get a `RequestLimitExceededException`. We could extend the `RequestLimitExceededException` with the parameters given in the headers.

**Comment by sjmohanta** [comment]
Thanks for the reply.

> You have exceeded the maximum number of request allowed by TMDb please try again later

Interestingly i have never got the above error message in my exception logs while checking for the error so far.

I don't have any new error logs since yesterday (when i cleared the error logs). I will post the error details once i get the issue again.

**Comment by LordMike** [comment]
Hmm. Maybe you (or we) have something that swallows it?

I can see something else wrong though. We should keep the HttpClient object for longer periods than one request, to make use of its features reg. keeping HTTP connections alive.

**Comment by sjmohanta** [comment]
Got the error again today.

Message: The method or operation is not implemented.
Stack Trace:

> at TMDbLib.Client.TMDbClient.get_RetryWaitTimeInSeconds()     at Tvsi.Business.Admin.Api.TMDBMovie.<DetailsByImdbIdAsync>d__2.MoveNext()  --- End of stack trace from previous location where exception was thrown ---     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)     at

It worked again after some time though but no message indicating request limit exceed.

I am using it like

`var res = await client.GetMovieAsync(imdbId: imdbId, extraMethods: TmDbObj.MovieMethods.Credits)`;

Which is called from action method. Am i missing something?

**Comment by LordMike** [comment]
Are you using an old version of the library?

https://github.com/LordMike/TMDbLib/blob/37ceaab6b75d2189d80f4a0944061b83cfdc1bb1/TMDbLib/Client/TMDbClient.cs#L92-L97

This code is obsolete. It should be removed, but it is not used inside the library. If you're calling the `RetryWaitTimeInSeconds` property anywhere in your code, that is where the bug is.

**Comment by sjmohanta** [comment]
> This code is obsolete. It should be removed, but it is not used inside the library. If you're calling the RetryWaitTimeInSeconds property anywhere in your code, that is where the bug is.

Yes, this was the issue. I was trying to read wait time on error.

**Comment by LordMike** [comment]
Heh. I've removed the code now (pending merge). Once merged I'll release 1.2.x (because, API change). It also exposes the retry-after values inside the retry exception.
